Paper: Semi-Supervised Semantic Annotator (S3A): Toward Efficient Semantic Image Labeling (#693)
scipy-conference/scipy_proceedings: papers/221_jessurun/sections_rst/intro.rst

Introduction
============

> Labeled image data is essential for tuning and evaluating the performance of machine learning applications.

Reviewer: ```suggestion
Labeled image data is essential for training, tuning, and evaluating the performance of many machine learning applications.
```

I recommend adding "training" to the list of tasks, and changing "machine learning applications" to "many machine learning applications". As written, this sentence makes it sound like this paper is relevant to every machine learning project, which is not true...a wide range of machine learning tasks do not rely on image data at all, such as regression/ranking/classification using tabular data.

> Such labels are typically defined with approximate enclosing shapes (i.e. simple polygons or parametric shapes), which tend to misrepresent more complex components.

Reviewer: ```suggestion
Such labels are typically defined with approximate enclosing shapes (e.g. "this square contains a cat" ), which tend to misrepresent more complex components (e.g. "the cat only occupies 70% of the square").
```

I suspect that a wide, interdisciplinary audience might not know what "parametric shapes" means. Reading this paper is the first time I'd ever heard that phrase.

Per https://github.com/scipy-conference/scipy_proceedings/blob/master/review_criteria.md#written-quality

> highly domain-specific jargon should be avoided or explained where possible

Would you consider changing this sentence to something like what I've suggested above, with plain-language descriptions of the topic? I think it might help a wider audience to understand the importance of what is being proposed.

When high accuracy is required, labels must be specified at or close to the pixel-level - a process known as semantic labeling or semantic segmentation.
A detailed description of this process is given in :cite:`chengSurveyAnalysisAutomatic2018`. Examples can readily be found in several popular datasets such as COCO, depicted in Figure :ref:`figsampleSegData`.

.. raw:: latex

    \makeSampleSegFig

..
    % However, a wide variety of applications require pixel-level accuracy, ranging from hardware assurance to medical imaging.

..
    % Alternatively, semantic segmentation is a technique providing pixel-level accuracy which avoids poor foreground representation :cite:`chengSurveyAnalysisAutomatic2018`.

..

>     %One such field applying this method include Bill-of-Material (BoM) extraction. This BoM, or list of all surface-mount devices (SMDs) on a printed circuit board (PCB) surface, can be generated from optical images of a board under test. Next, it can be compared against a reference design to detect likely counterfeit, tampered, or defective SMDs :cite:`paradis2020color,azhaganReviewAutomaticBill2019`. To create such a BoM using optical data alone, it is crucial that detected SMDs use segmentation masks that are representative of structures such as pins, solder flow, and more. This is the only way the collected data will remain indicative of true BoM properties. An example annotation is shown in Figure :ref:`figpcb`.

Reviewer: ```suggestion
    %One such field applying this method includes Bill-of-Material (BoM) extraction. This BoM, or list of all surface-mount devices (SMDs) on a printed circuit board (PCB) surface, can be generated from optical images of a board under test. Next, it can be compared against a reference design to detect likely counterfeit, tampered, or defective SMDs :cite:`paradis2020color,azhaganReviewAutomaticBill2019`. To create such a BoM using optical data alone, it is crucial that detected SMDs use segmentation masks that are representative of structures such as pins, solder flow, and more. This is the only way the collected data will remain indicative of true BoM properties. An example annotation is shown in Figure :ref:`figpcb`.
```

I've suggested one typo change here. It would be easier to see if these `.rst` files were reformatted following the suggestions in https://github.com/scipy-conference/scipy_proceedings/blob/master/review_criteria.md#other-requirements

> Avoid long paragraphs of your rich text file to enable reviewers to target specific sections with their comments. GitHub comments are per line of text.

Reviewer: ^ and I should have included in my review summary...this suggestion to break multi-sentence paragraphs into individual lines, for purpose of improving the review process, extends to every file in this pull request

Author: Paragraphs split with newlines per recommendation

In all these cases, imprecise annotations severely limit the development of automated solutions and can decrease the accuracy of standard trained segmentation models.

Quality semantic segmentation is difficult due to a reliance on large, high-quality datasets, which are often created by manually labeling each image.
Manual annotation is error-prone, costly, and greatly hinders scalability. As such, several tools have been proposed to alleviate the burden of collecting these ground-truth labels :cite:`BestImageAnnotation`.
Unfortunately, existing tools are heavily biased toward lower-resolution images with few regions of interest (ROI), similar to Figure :ref:`figsampleSegData`.
While this may not be an issue for some datasets, such assumptions are *crippling* for high-fidelity images with hundreds of annotated ROIs :cite:`Ladicky_whatWhereCombiningCRFs,Wang_multiLabelImageAnnotation`.

..
    % This scenario is represented in Figure :ref:`figbees` but can occur in multiple of the domains previously listed.

..

>     % Especially when each region can be arbitrarily complex, the software will enter a non-responsive state where no annotation can be performed.

Reviewer: ```suggestion
    % Especially when each region can be arbitrarily complex, the software can enter a non-responsive state where no annotation can be performed.
```
